File: 10_laba/src/by/bntu/fitr/java/inGoslingsFootsteps/Lab10/model/entity/Flight.java
package by.bntu.fitr.java.inGoslingsFootsteps.Lab10.model.entity;

import java.util.Arrays;

public class Flight {
    private final static int MIN_COUNT_OF_TICKETS = 400;
    private final static int MIN_PRICE= 100;

    private String departurePoint;
    private String destination;
    private String departureDate;
    private String departureTime;
    private int countOfTicket;
    private int price;
    private String[] customersId;

    public Flight() {
        departurePoint = "no name";
        destination = "no name";
        departureDate = "no name ";
        departureTime = "no name";
        countOfTicket = MIN_COUNT_OF_TICKETS;
        price = MIN_PRICE;
    }

    public Flight(String departurePoint,String destination, String departureDate, String departureTime, int countOfTicket, int price) {
        this.departurePoint = departurePoint;
        this.destination = destination;
        this.departureDate = departureDate;
        this.departureTime = departureTime;
        this.countOfTicket = countOfTicket;
        this.price = price;
    }

    public Flight(Flight flight) {
        this(flight.departurePoint, flight.destination, flight.departureDate, flight.departureTime, flight.countOfTicket, flight.price);
    }

    public String getDeparturePoint() {
        return departurePoint;
    }

    public void setDeparturePoint(String departurePoint) {
        this.departurePoint = departurePoint;
    }

    public String getDestination() {
        return destination;
    }

    public void setDestination(String destination) {
        this.destination = destination;
    }

    public String getDepartureDate() {
        return departureDate;
    }

    public void setDepartureDate(String departureDate) {
        this.departureDate = departureDate;
    }

    public String getDepartureTime() {
        return departureTime;
    }

    public void setDepartureTime(String departureTime) {
        this.departureTime = departureTime;
    }

    public int getCountOfTicket() {
        return countOfTicket;
    }

    public void setCountOfTicket(int countOfTicket) {
        this.countOfTicket = countOfTicket;
    }

    public int getPrice() {
        return price;
    }

    public void setPrice(int price) {
        this.price = price;
    }

    public String[] getCustomersId() {
        return customersId;
    }

    public void setCustomersId(String[] customersId) {
        this.customersId = customersId;
    }

    public void addCustomersId(Customer customer) {
        this.customersId = Arrays.copyOf(this.customersId, this.customersId.length + 1);
        this.customersId[this.customersId.length - 1] = customer.getId();
    }

    public void  removeCustomerId(int index) {
        int size = this.customersId.length;
        String[] newCustomerId = new String[size - 1];
        System.arraycopy(this.customersId, 0, newCustomerId, 0, index);
        System.arraycopy(this.customersId, index + 1, newCustomerId, index,
                size - index - 1);
        this.customersId = new String[newCustomerId.length];
        System.arraycopy(newCustomerId, 0, this.customersId, 0, newCustomerId.length);
    }

    @Override
    public String toString() {
        return departurePoint  + " ---> " + destination + ", Date - " + departureDate + ", Time - " + departureTime
                + ", Price - " + price;
    }
}

File: 10_laba/C##/AirTicketOffice/AirTicketOffice/by/bntu/fitr/java/ingoslingsfootsteps/model/entity/Flight.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace AirTicketOffice.by.bntu.fitr.java.ingoslingsfootsteps.model.entity
{
    public class Flight
    {
        private const int MIN_COUNT_OF_TICKETS = 400;
        private const int MIN_PRICE = 100;

        private string departurePoint;
        private string destination;
        private string departureDate;
        private string departureTime;
        private int countOfTicket;
        private int price;
        private string[] customersId;

        public Flight()
        {
            departurePoint = "no name";
            destination = "no name";
            departureDate = "no name ";
            departureTime = "no name";
            countOfTicket = MIN_COUNT_OF_TICKETS;
            price = MIN_PRICE;
        }

        public Flight(String departurePoint, String destination, String departureDate, String departureTime, int countOfTicket, int price)
        {
            this.departurePoint = departurePoint;
            this.destination = destination;
            this.departureDate = departureDate;
            this.departureTime = departureTime;
            this.countOfTicket = countOfTicket;
            this.price = price;
        }

        public Flight(Flight flight) : this(flight.departurePoint, flight.destination, flight.departureDate, flight.departureTime, flight.countOfTicket, flight.price)
        {
        }

        public string DeparturePointame
        {
            get
            {
                return this.departurePoint;
            }
            set
            {
                this.departurePoint = value;
            }
        }

        public string Destination
        {
            get
            {
                return this.destination;
            }
            set
            {
                this.destination = value;
            }
        }

        public string DepartureDate
        {
            get
            {
                return this.departureDate;
            }
            set
            {
                this.departureDate = value;
            }
        }

        public string DepartureTime
        {
            get
            {
                return this.departureTime;
            }
            set
            {
                this.departureTime = value;
            }
        }

        public int CountOfTicket
        {
            get
            {
                return this.countOfTicket;
            }
            set
            {
                this.countOfTicket = value;
            }
        }

        public int Price
        {
            get
            {
                return this.price;
            }
            set
            {
                this.price = value;
            }
        }

        public string[] CustomersId
        {
            get
            {
                return this.customersId;
            }
            set
            {
                this.customersId = value;
            }
        }

        public void AddCustomersId(Customer customer)
        {
            Array.Resize(ref this.customersId, this.customersId.Length + 1);
            this.customersId[this.customersId.Length - 1] = customer.Id;
        }

        public void RemoveCustomerId(int index)
        {
            int size = this.customersId.Length;
            String[] newCustomersEmail = new String[size - 1];
            System.Array.Copy(this.customersId, 0, newCustomersEmail, 0, index);
            System.Array.Copy(this.customersId, index + 1, newCustomersEmail, index,
                    size - index - 1);
            this.customersId = new String[newCustomersEmail.Length];
            System.Array.Copy(newCustomersEmail, 0, this.customersId, 0, newCustomersEmail.Length);
        }

        public override string ToString()
        {
            return departurePoint + " ---> " + destination + ", Date - " + departureDate + ", Time - " + departureTime
                    + ", Price - " + price;
        }
    }
}
